manager/src/main/java/org/openremote/manager/rules/RulesEngine.java

```diff
@@ -148,6 +148,7 @@ public RulesEngine(TimerService timerService,
                        MeterRegistry meterRegistry) {
         this.timerService = timerService;
         this.rulesService = rulesService;
+        this.previouslyFired = rulesService.startDone;
```

Comment:
When a rule engine is instantiated then `previouslyFired` should be false. A rule engine can be created after the system has started (e.g. a new realm is added, or the first rule is added to a realm).

Only once rule engine has fired all deployments once should this flag be updated...I don't see anywhere that this flag gets updated anymore.

Reply:
The issue is that it doesn't fire if there are no rules in a realm because then there is no `RulesEngine`. It only creates a `RulesEngine` and starts it after you create a new rule when there are no rules. That was the reason why the tests failed because they added rules to a running instance in a realm that did not have any rules. It can have a better name though. :wink:

Reply:
How about also naming it `startDone`?